api/client/trust.go

```diff
@@ -0,0 +1,445 @@
+package client
+
+import (
+	"bufio"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/distribution/digest"
+	"github.com/docker/distribution/registry/client/auth"
+	"github.com/docker/distribution/registry/client/transport"
+	"github.com/docker/docker/cliconfig"
+	"github.com/docker/docker/pkg/ansiescape"
+	"github.com/docker/docker/pkg/ioutils"
+	flag "github.com/docker/docker/pkg/mflag"
+	"github.com/docker/docker/pkg/tlsconfig"
+	"github.com/docker/docker/registry"
+	"github.com/docker/notary/client"
+	"github.com/docker/notary/pkg/passphrase"
+	"github.com/docker/notary/trustmanager"
+	"github.com/endophage/gotuf/data"
+)
+
+var untrusted bool
+
+func addTrustedFlags(fs *flag.FlagSet, verify bool) {
+	var trusted bool
+	if e := os.Getenv("DOCKER_TRUST"); e != "" {
+		var err error
+		trusted, err = strconv.ParseBool(e)
+		if err != nil {
+			// treat any other value as true
+			trusted = true
+		}
+	}
+	message := "Skip image signing"
+	if verify {
+		message = "Skip image verification"
+	}
+	fs.BoolVar(&untrusted, []string{"-untrusted"}, !trusted, message)
+}
+
+func isTrusted() bool {
+	// TODO(dmcgowan): false if name is hex image id
+	// TODO(dmcgowan): whitelist specific names to be untrusted
+	// TODO(dmcgowan): canonicalize name before checking whitelist
+	return !untrusted
```

Comment:
As soon as 1.8.0 is out and we announce image signature, people will want to look at its implementation. This is not giving a good image: are those TODO features/optims that will make it in the future? Would they be better filed as issues?

I'm fine documenting potential future improvements as comments in the code, but in that let's explain clearly what they mean, or remove them altogether. Here it's vague enough to become suspicious.